### PR TITLE
Expose shape_op python api

### PR DESCRIPTION
**Pull request by baiyfbupt**
It will be used in Detection API to support variable length data inference.

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -5354,3 +5355,37 @@ def rank_loss(label, left, right, name=None):
...
+
+
+def shape(input, name=None):
+    """
+    **Shape layer**
+   
+    Get the shape of input tensor.
+
+    Args:
+        input(Variable): An input tensor.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: add `name` in Args.

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -5354,3 +5355,37 @@ def rank_loss(label, left, right, name=None):
...
+    
+    Returns:
+        Variable: Shape of the input tensor.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: 需要给出返回的形状

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -5354,3 +5355,37 @@ def rank_loss(label, left, right, name=None):
...
+    
+    Raises:
+        ValueError: If input is not a Variable.
+
+    Examples:
+        
+        .. code-block:: python
+
+            data = fluid.layers.data(name="data", shape =(3,100,100), dtype="float32")
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: shape=(3,100,100)

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -5354,3 +5355,37 @@ def rank_loss(label, left, right, name=None):
                 "Right": right},
         outputs={'Out': out})
     return out
+
+
+def shape(input, name=None):
+    """
+    **Shape layer**
+   
+    Get the shape of input tensor.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py:5364]: 可以描述下，比如input的shape是[2, 4, 8]，返回值是一个rank为1的tensor，值是(2,4,8)，这个tensor的类型是 `int64_t`

Comment (baiyfbupt) [reply python/paddle/fluid/layers/nn.py:5364]: Done

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/nn.py]: Done

**Comment by qingqing01** [comment]
```
[12:55:38][Step 1/1] API Difference is: 
[12:55:38][Step 1/1] + paddle.fluid.layers.rank_loss ArgSpec(args=['label', 'left', 'right', 'name'], varargs=None, keywords=None, defaults=(None,))
[12:55:38][Step 1/1] + paddle.fluid.layers.shape ArgSpec(args=['input', 'name'], varargs=None, keywords=None, defaults=(None,))
[12:55:38][Step 1/1] - paddle.fluid.layers.shape ArgSpec(args=[], varargs='args', keywords='kwargs', defaults=None)
[12:55:39][Step 1/1] Process exited with code 1
[12:55:39][Step 1/1] Process exited with code 1
[12:55:39][Step 1/1] Step Build and test (Command Line) failed
```

Need to update the `paddle/fluid/API.spec`

**Pull request closed without merging**